File: 2023-2024/sourcecode/mod1/lab2/42DynamicFragmentBasicsJava/app/src/main/java/com/example/a42_dynamicfragmentbasics_java/MainFragment.java
package com.example.a42_dynamicfragmentbasics_java;

import android.os.Bundle;

import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.a42_dynamicfragmentbasics_java.databinding.FragmentMainBinding;

public class MainFragment extends Fragment {

    private FragmentMainBinding binding;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        binding = FragmentMainBinding.inflate(inflater);

        binding.startFragmentA
                .setOnClickListener(v -> requireActivity().getSupportFragmentManager()
                        .beginTransaction()
                        .replace(R.id.fragment_container_view_tag, new FragmentA())
                        .addToBackStack(null)
                        .commit());

        return binding.getRoot();
    }
}

File: 2023-2024/sourcecode/mod1a/lab2/41DynamicFragmentBasicsKotlin/app/src/main/java/com/example/a41_dynamicfragmentbasics_kotlin/MainFragment.kt
package com.example.a41_dynamicfragmentbasics_kotlin

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.example.a41_dynamicfragmentbasics_kotlin.databinding.FragmentMainBinding

class MainFragment : Fragment() {

    private lateinit var binding: FragmentMainBinding

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        binding = FragmentMainBinding.inflate(inflater)

        binding.startFragmentA.setOnClickListener {
            requireActivity().supportFragmentManager
                .beginTransaction()
                .replace(R.id.fragment_container_view_tag, FragmentA())
                .addToBackStack(null)
                .commit()
        }

        return binding.root
    }
}